src/components/usePopup.ts

```diff
@@ -1,18 +1,83 @@
 import { useEffect, useState } from "react";
-import { CountryName } from "../constants/Countries";
+import { Countries, CountryName, WorldAverage } from "../constants/Countries";
 
 export type PopupProps = {
-    selectedCountries: Set<CountryName>;
+    selectedCountries: Map<CountryName, number>;
     addSelectedCountry: (country: CountryName) => void;
     removeSelectedCountry: (country: CountryName) => void;
-    refreshAndGetSize: (selectedCountry: Set<CountryName>) => Promise<void>;
+    setCountryPercentage: (country: CountryName, percentage: number) => void;
+    averageSpecificEmissions: number;
+    refreshAndGetSize: () => Promise<void>;
+    error: string | null;
```

Comment:
can this be error?: string or does it require null?

Reply:
Just realised that you can assign undefined to error as well. 

I guess it works in both cases, should I use undefineable or nullable, i.e.
- error?: string OR
- error: string | null

I've seen documentation [encouraging null ](https://dev.to/typescripttv/what-is-the-difference-between-null-and-undefined-5h76#:~:text=The%20convention%20in%20TypeScript%20is,intentional%20absence%20of%20a%20value.) and also others [encouraging undefined](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined)

Reply:
Personally I prefer undefined, I think the main job is to stay consistent and not use both all around the code base?